### PR TITLE
Homework5(tree)

### PR DESCRIPTION
**Pull request by cAfton**

### CODE REVIEW

File: homeworks/Homework5/TreeDAI/TreeDAI.cpp [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+using namespace std;
+
+enum Offenses
+{
+	speeding,
+	illegal_parking,
+	running_a_red_light,
+	no_seatbelt,
+	phone_use_while_driving,
+};
```
Comment (Yurii-Vipshovsky) [inline homeworks/Homework5/TreeDAI/TreeDAI.cpp:12]: oo cool!

File: homeworks/Homework5/TreeDAI/TreeDAI.cpp [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+	Car(int number, int size, Offenses* offense) {
+		this->carNumber = number;
+		this->size = size;
+		this->offense = offense;
+	}
+
+	Car(int number) {
+		this->carNumber = number;
+		this->size = 0;
+		this->offense = new Offenses[size];
```
Comment (Yurii-Vipshovsky) [inline homeworks/Homework5/TreeDAI/TreeDAI.cpp:35]: it's not logick to create an array with size 0!

File: homeworks/Homework5/TreeDAI/TreeDAI.cpp [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+	int getNumber() {
+		return this->carNumber;
+	}
+
+	int getSize() {
+		return this->size;
+	}
+
+	void addOffense(Offenses newOffense) {
+		Offenses* temp = new Offenses[size + 1];
```
Comment (Yurii-Vipshovsky) [inline homeworks/Homework5/TreeDAI/TreeDAI.cpp:48]: Can be optimized, not to create a ne array any time

File: homeworks/Homework5/TreeDAI/TreeDAI.cpp [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+		offense = temp;
+		size++;
+	}
+
+	friend ostream& operator<<(ostream& out, Car car) {
+		out << "\nCar number: " << car.carNumber << endl;
+		out << "Offenses: \n";
+		for (size_t i = 0; i < car.size; i++)
+		{
+			switch (car.offense[i])
```
Comment (Yurii-Vipshovsky) [inline homeworks/Homework5/TreeDAI/TreeDAI.cpp:64]: I think it also can be siplifized, but not shure, tranform enum without switch
Try to google

File: homeworks/Homework5/TreeDAI/TreeDAI.cpp [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+					temp.addOffense(running_a_red_light);
+					break;
+				case 4:
+					temp.addOffense(no_seatbelt);
+					break;
+				case 5:
+					temp.addOffense(phone_use_while_driving);
+					break;
+				default:
+					cout << "wrong choose" << endl;
```
Comment (Yurii-Vipshovsky) [inline homeworks/Homework5/TreeDAI/TreeDAI.cpp:280]: You write this code 2 times, think about how this try catch can be simplified

File: homeworks/Homework5/TreeDAI/TreeDAI.cpp [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+		else if (userChoise == 2) {
+			cout << DAI << endl;
+			system("pause");
+		}
+		else if (userChoise == 3) {
+			int numFind;
+			cout << "enter number to find: ";
+			cin >> numFind;
+			system("cls");
+			try {
```
Comment (Yurii-Vipshovsky) [inline homeworks/Homework5/TreeDAI/TreeDAI.cpp:327]: take a look here!

**Review by Yurii-Vipshovsky — COMMENTED** [review]
Generally good but can be improved

File: homeworks/Homework5/TreeDAI/TreeDAI.cpp (hunk 4 above)
Comment (cAfton) [reply homeworks/Homework5/TreeDAI/TreeDAI.cpp:64]: didn't found(

**Pull request merged**